### PR TITLE
Remove JUNIT contract tests and solve JPA issues

### PR DESCRIPTION
**Pull request by chibenwa**
Also added one test on duplicated mailbox names at the mapper level.

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

File: mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/JPAMailboxManagerTest.java [hunk 1]
```diff
@@ -43,10 +42,7 @@ protected MailboxManager provideMailboxManager() {
 
     @After
     public void tearDown() throws MailboxException {
-        if (openJPAMailboxManager.isPresent()) {
-            openJPAMailboxManager.get()
-                .deleteEverything(new MockMailboxSession("Any name"));
-        }
+        JPA_TEST_CLUSTER.clear(JPAMailboxFixture.MAILBOX_TABLE_NAMES);
```
Comment (mbaechler) [inline mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/JPAMailboxManagerTest.java]: this "CLUSTER" thing becomes ridiculous

File: backends-common/jpa/src/test/java/org/apache/james/backends/jpa/JpaTestCluster.java [hunk 2]
```diff
@@ -84,7 +85,7 @@ public void clear(List<String> tables) {
         EntityManager entityManager = entityManagerFactory.createEntityManager();
         entityManager.getTransaction().begin();
         for(String tableName: tables) {
-            entityManager.createNativeQuery("TRUNCATE table " + tableName).executeUpdate();
+            entityManager.createNativeQuery("DELETE FROM " + tableName).executeUpdate();
```
Comment (mbaechler) [inline backends-common/jpa/src/test/java/org/apache/james/backends/jpa/JpaTestCluster.java]: I really prefer using a new database everytime, probably much faster, too

File: mailbox/api/pom.xml [hunk 3]
```diff
@@ -41,12 +41,17 @@
...
+            <artifactId>commons-codec</artifactId>
+        </dependency>
+        <dependency>
             <groupId>commons-lang</groupId>
             <artifactId>commons-lang</artifactId>
         </dependency>
         <dependency>
-            <groupId>commons-codec</groupId>
-            <artifactId>commons-codec</artifactId>
+            <groupId>commons-io</groupId>
```
Comment (strokyl) [inline mailbox/api/pom.xml]: Weird diff to sum up you had commons-io in test why do you need it ?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerProvider.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
+/****************************************************************
```
Comment (strokyl) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerProvider.java]: You commit say that you just remove all junit test and here you add this class why ?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
+/****************************************************************
```
Comment (strokyl) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java]: Same here, this should not be in your commit unless I misunderstood something

File: backends-common/jpa/src/test/java/org/apache/james/backends/jpa/JpaTestCluster.java (hunk 2 above)
Comment (aduprat) [inline backends-common/jpa/src/test/java/org/apache/james/backends/jpa/JpaTestCluster.java]: Yes, but not easy without breaking everything...

File: mailbox/api/pom.xml (hunk 3 above)
Comment (aduprat) [inline mailbox/api/pom.xml]: We need it in *AttachmentTests* (IOUtils).

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java (hunk 5 above)
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java]: Sorry, but I don't understand your comment.

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Pull request merged**